### PR TITLE
Add React template for contacts

### PR DESCRIPTION
**Pull request by cgsunkel**
## Description of change

The old Nunjucks template for the contacts pages has been replaced by a new React one.
## Test instructions

All contact pages should render as normal.

### Before

Screenshot 2023-02-06 at 13 39 42

### After

Screenshot 2023-02-06 at 13 38 54

## Checklist

[//]: # 'When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/main/docs/Code%20review%20guidelines.md'

- [ ] Has the branch been rebased to main?
- [ ] Automated tests (Any of the following when applicable: Unit, Functional or End-to-End)
- [ ] Manual compatibility testing (Browsers: Chrome, Firefox, Edge, Safari)

### CODE REVIEW

File: src/apps/interactions/controllers/list.js [hunk 1]
```diff
@@ -122,9 +119,22 @@ function renderInteractionsForEntity(req, res, next) {
 
     res.breadcrumb(breadcrumbTitle).render(view, {
       actionButtons,
+    })
+  } catch (error) {
+    next(error)
+  }
+}
+
+function renderContactActivityForEntity(req, res, next) {
```
Comment (cgsunkel) [inline src/apps/interactions/controllers/list.js:128]: This has been split into two as these two pages have significantly diverged and adding further items to the shared controller was causing issues with the investments interaction list

File: src/apps/contacts/controllers/activity.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+function renderContactActivityForEntity(req, res, next) {
```
Comment (cgsunkel) [inline src/apps/contacts/controllers/activity.js:1]: This has been split into two as these two pages have significantly diverged and adding further items to the shared controller was causing issues with the investments interaction list

File: src/apps/contacts/controllers/audit.js [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
   try {
     const contactId = req.params.contactId
-    const contact = res.locals?.contact
+    const permissions = res.locals.user.permissions
 
     return res.breadcrumb('Audit history').render('contacts/views/audit', {
       props: {
         contactId: contactId,
-        contact: contact,
+        permissions: permissions,
```
Comment (paulgain) [inline src/apps/contacts/controllers/audit.js:9]: You can shorten this to `permissions,`

File: src/apps/contacts/controllers/details.js [hunk 4]
```diff
@@ -30,14 +30,14 @@ async function getCommon(req, res, next) {
...
-    const contact = res.locals.contact
     const companyAddress = res.locals.company.address
+    const permissions = res.locals.user.permissions
 
     res.render('contacts/views/details', {
       props: {
         contactId: contactId,
-        contact: contact,
         companyAddress: companyAddress,
+        permissions: permissions,
```
Comment (paulgain) [inline src/apps/contacts/controllers/details.js:40]: Ditto!

File: src/client/components/Layout/ContactLayout.jsx [hunk 5]
```diff
@@ -0,0 +1,75 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import GridCol from '@govuk-react/grid-col'
+import GridRow from '@govuk-react/grid-row'
+import { SPACING } from '@govuk-react/constants'
+
+import { Main } from '../../components'
+import { ContactLocalHeader, LocalNav, LocalNavLink } from '../../components'
```
Comment (paulgain) [inline src/client/components/Layout/ContactLayout.jsx:9]: No need for the double import.
```js
import { Main, ContactLocalHeader, LocalNav, LocalNavLink } from '../../components'
```

File: src/client/modules/Contacts/ContactDetails/ContactDetails.jsx [hunk 6]
```diff
@@ -43,81 +44,86 @@ const getAddress = (contact, companyAddress) => {
   return Object.values(addressCleaned).join(', ')
 }
 
-const ContactDetails = ({ contactId, companyAddress }) => {
+const ContactDetails = ({ contactId, companyAddress, permissions }) => {
   return (
```
Comment (paulgain) [inline src/client/modules/Contacts/ContactDetails/ContactDetails.jsx:48]: Favour implicit returns over explicit returns.

**Review by paulgain — APPROVED** [review]
Looks good, just a few minor comments.

File: src/apps/contacts/controllers/audit.js [hunk 7]
```diff
@@ -1,12 +1,12 @@
 async function getAudit(req, res, next) {
   try {
     const contactId = req.params.contactId
-    const contact = res.locals?.contact
+    const permissions = res.locals.user.permissions
```
Comment (simon-mt) [inline src/apps/contacts/controllers/audit.js:4]: Is there an opportunity for the local object to be null here, and throw an exception? Just looking at the previous line 4 and wondered if the ? was there for a specific reason.

**Review by simon-mt — CHANGES_REQUESTED** [review]

**Review by simon-mt — APPROVED** [review]
